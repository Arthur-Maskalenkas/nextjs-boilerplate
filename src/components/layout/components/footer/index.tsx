import React from 'react'

import styles from 'components/layout/components/footer/styles.module.scss'

export const Footer = () => {
  return (
    <footer data-testid="main-footer" className={`${styles.container} `}>
      Footer
    </footer>
  )
}