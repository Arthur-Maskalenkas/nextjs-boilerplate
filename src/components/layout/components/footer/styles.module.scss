@use 'styles/main';

$variants {

}

.container {
  @extend %variants;

  height: main.$height-container-footer-mobile;

  @include main.media('>=#{main.$media-desktop-1024}') {
    height: main.$height-container-footer-desktop;
  }
}
