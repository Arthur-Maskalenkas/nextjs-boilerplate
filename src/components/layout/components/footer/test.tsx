import React from 'react'
import { render, screen, RenderResult } from '@testing-library/react'

import { Footer } from 'components/layout/components/footer/index'

type SutTypes = {
	sut: RenderResult
}

const makeSut = (): SutTypes => {
  const sut = render(<Footer />)

  return {
    sut
  }
}

describe('Footer', () => {
  test('Should render Footer correctly', () => {
    const { sut: { container } } = makeSut()

    expect(screen.getByText(/Footer/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
