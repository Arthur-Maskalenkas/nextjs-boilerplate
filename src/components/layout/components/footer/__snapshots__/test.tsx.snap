// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render Footer correctly 1`] = `
<div>
  <footer
    class="container "
    data-testid="main-footer"
  >
    Footer
  </footer>
</div>
`;
